src/inmanta/app.py

```diff
@@ -620,6 +620,74 @@ def export(options: argparse.Namespace) -> None:
     summary_reporter.print_summary_and_exit(show_stack_traces=options.errors)
 
 
+def validate_logging_config_parser_config(
+    parser: argparse.ArgumentParser, parent_parsers: abc.Sequence[ArgumentParser]
+) -> None:
+    """
+    Config parser for the validate-logging-config command.
+    """
+    parser.add_argument(
+        "-e",
+        dest="environment",
+        help="The environment id to be used as context variable in logging config templates. If not specified,"
+        " 0c111d30-feaf-4f5b-b2d6-83d589480a4a will be used.",
+        default="0c111d30-feaf-4f5b-b2d6-83d589480a4a",
+    )
+
+    sub_parsers = parser.add_subparsers(title="subcommand", dest="cmd")
+    for component_name in ["server", "scheduler", "compiler"]:
+        sub_parser = sub_parsers.add_parser(
+            component_name, help=f"Validate the logging config for the {component_name}", parents=parent_parsers
+        )
+        sub_parser.set_defaults(component=component_name)
+
+
+@command(
+    "validate-logging-config",
+    help_msg="This command loads a logging config file, taking into account the precedence rules for logging config files,"
+    " and produces log lines. It serves as a tool to validate whether a logging config file is syntactically correct"
+    " and behaves as expected. Optionally, a sub-command can be specified to indicate the component for which the"
+    " logging config file should be loaded.",
+    parser_config=validate_logging_config_parser_config,
+)
+def validate_logging_config(options: argparse.Namespace) -> None:
+    logging_config = InmantaLoggerConfig.get_current_instance()
```

Comment:
Doesn't this raise an exception if the config isn't loaded yet?

Reply:
Yes. But the config is loaded in the `app.app()` function before it calls into this method.